### PR TITLE
Fix comments for #1890

### PR DESCRIPTION
**Pull request by junderw**
Fixing some comments that were left on #1890

### CODE REVIEW

File: bitcoin/src/blockdata/script/borrowed.rs [hunk 1]
```diff
@@ -354,8 +360,8 @@ impl Script {
         let mut n = 0;
         let mut pushnum_cache = None;
         for inst in self.instructions() {
-            match inst? {
```
Comment (Kixunil) [inline bitcoin/src/blockdata/script/borrowed.rs]: Why not use question mark? Seemed fine to me.

File: bitcoin/src/blockdata/opcodes.rs [hunk 2]
```diff
@@ -422,9 +422,10 @@ impl All {
     ///
     /// # Examples
     ///
-    /// ```
+    /// ```no_run
```
Comment (Kixunil) [inline bitcoin/src/blockdata/opcodes.rs:425]: Why?

Comment (junderw) [reply bitcoin/src/blockdata/opcodes.rs:425]: It's not public, so it doesn't compile.

Comment (junderw) [reply bitcoin/src/blockdata/opcodes.rs]: I got it confused with `ignore`

File: bitcoin/src/blockdata/script/borrowed.rs (hunk 1 above)
Comment (junderw) [reply bitcoin/src/blockdata/script/borrowed.rs]: Either way is fine. Reverted.

File: bitcoin/src/blockdata/script/borrowed.rs [hunk 3]
```diff
@@ -332,6 +332,9 @@ impl Script {
     /// (Note: taproot scripts don't count toward the sigop count of the block,
     /// nor do they have CHECKMULTISIG operations. This function does not count OP_CHECKSIGADD,
     /// so do not use this to try and estimate if a taproot script goes over the sigop budget.)
+    ///
+    /// # Errors
+    /// If `self.instructions()` encounters an error during iteration, that error is returned.
```
[the remark this answers is not in the document]
Comment (junderw) [reply bitcoin/src/blockdata/script/borrowed.rs]: Valid-Invalid seems too loaded for what instructions error is.

I rewrote as "If the Script is not able to be parsed to completion."

File: bitcoin/src/blockdata/opcodes.rs [hunk 4]
```diff
@@ -422,12 +422,13 @@ impl All {
     ///
     /// # Examples
     ///
-    /// ```
+    /// ```ignore
     /// use bitcoin::opcodes::all::*;
-    /// assert_eq!(OP_PUSHNUM_5.decode_pushnum().expect("pushnum"), 5)
+    /// assert_eq!(OP_PUSHNUM_5.decode_pushnum().expect("pushnum"), 5);
+    /// assert!(OP_DROP.decode_pushnum().is_none());
```
Comment (tcharding) [inline bitcoin/src/blockdata/opcodes.rs]: This doesn't work as a test if its ignored. Can you move it to unit tests?

File: bitcoin/src/blockdata/script/borrowed.rs [hunk 5]
```diff
@@ -346,6 +349,9 @@ impl Script {
     /// (Note: taproot scripts don't count toward the sigop count of the block,
     /// nor do they have CHECKMULTISIG operations. This function does not count OP_CHECKSIGADD,
     /// so do not use this to try and estimate if a taproot script goes over the sigop budget.)
+    ///
+    /// # Errors
+    /// If the Script is not able to be parsed to completion.
```
Comment (tcharding) [inline bitcoin/src/blockdata/script/borrowed.rs:354]: nit: Could have newline between heading and text.

File: bitcoin/src/blockdata/opcodes.rs [hunk 6]
```diff
@@ -577,6 +570,12 @@ mod tests {
         assert_eq!(s, "    OP_NOP");
     }
 
+    #[test]
+    fn decode_pushnum() {
+        assert_eq!(OP_PUSHNUM_5.decode_pushnum().expect("pushnum"), 5);
+        assert!(OP_DROP.decode_pushnum().is_none());
+    }
```
Comment (Kixunil) [inline bitcoin/src/blockdata/opcodes.rs:577]: I think it'd be useful to test things around edges: one-below-min, min, max, one-above-max.

**Review by Kixunil — COMMENTED** [review]
Looks good

File: bitcoin/src/blockdata/opcodes.rs (hunk 6 above)
Comment (junderw) [reply bitcoin/src/blockdata/opcodes.rs:609]: Since there's only 16, I covered everything.

**Comment by junderw** [comment]
Rebased on master (the other PR got merged 10 minutes ago)

Ready for reviews.

**Review by apoelstra — APPROVED** [review]
ACK d961b9c4eeef02fa9c0f664e505e862bd366327b

**Review by tcharding — APPROVED** [review]
Nice! Thanks for doing the unit test fully.

ACK d961b9c4eeef02fa9c0f664e505e862bd366327b

**Pull request merged**